docs/ADR/2019-11-27-firestore-saving-timed-transcripts.md

```diff
@@ -0,0 +1,406 @@
+    },
+    {
+      "id": 1,
+      "start": 3.9,
+      "end": 4,
+      "text": "you"
+    },
+    {
+      "id": 2,
+      "start": 4,
+      "end": 4.1,
+      "text": "hear"
+    },
+    {
+      "id": 3,
+      "start": 4.1,
+      "end": 4.2,
+      "text": "it?"
+    },
+    ...
+```
+
+could be converted to one string, so a document in a collection
+```
+Text startTime endTime\ttext startTime endTime\t...
+```
+
+```
+Can 1.4 3.9\tyou 3.9 4\thear 4 4.1\tit? 4.1 4.2\t...
+```
+
+
+And then parse it back to a list of words objects.
+
+
+or it could be a collection with 3 documents. of type text string
+
+first one is text
+
+```
+Can\tyou\thear\tit?\t...
+```
+then a document for the Start timecodes
+
+```
+1.4\t3.9\t4\t...
+```
+
+last but not least a document for the End timecodes
+
+```
+3.9\t4\t4.2\t
+```
+
+eg quick example of code, for separate text, start, and end time.
+```js
+const dpeJson = require('./dpe.json');
+
+const text = dpeJson.words.map((word)=>{
+    return word.text;
+}).join('\t')
+
+console.log(text)
+
+const startTime = dpeJson.words.map((word)=>{
+    return word.start;
+}).join('\t')
+
+console.log(startTime)
+
+const endTime = dpeJson.words.map((word)=>{
+    return word.end;
+}).join('\t')
+
+console.log(endTime)
+```
+
+From 1.4M to 85K for end time, 85K for start time and 70K for text.
+
+Or combined
+
+```js
+const dpeJson = require('./dpe.json');
+
+console.log(dpeJson)
+
+const text = dpeJson.words.map((word)=>{
+    return `${word.text}\t${word.start}\t${word.end}`;
+}).join('\t')
+
+console.log(text)
+```
+example output (where the spaces are tabs `\t`)
+
+```js
+the     4511.7  4511.9  scrape  4511.9  4512.2  on      4512.2  4512.4  Ematic  4512.4  4512.6  Lee.    4512.6  4512.8  I       4514.6  4515.3  think   4515.3  4515.8  most    4515.8  4516.6  of      4516.6  4516.6  like    4516.6  4517.6  journalism      4517.6  4518    organizations.  4518    4518.4
+```
+
+From 1.4M `json` to 241K `tsv`.
+
+as a quick example of reassambly from tsv to json 
+
+```js
+// const fs = require('fs');
+//const textTsv  = fs.readFileSync('dpe-text.tsv').toString()
+// const startTimeTsv  = fs.readFileSync('dpe-startTime.tsv').toString()
+// const endTimeTimeTsv  = fs.readFileSync('dpe-startTime.tsv').toString()
+
+const startTimeList = startTimeTsv.split('\t');
+const endTimeList = endTimeTimeTsv.split('\t');
+
+const words = textTsv.split('\t').map((wordText, index)=>{
+    return {
+        text: wordText,
+        start: startTimeList[index],
+        end: endTimeList[index]
+    }
+})
+
+const result = { words} ;
+
+console.log(result);
+```
+
+Paragraphs would get saved in a similar way with speaker attribute instead of text attribute in a separate collection.
+
+The conversion could be done either on the client or in a cloud function.
+
+* Good, because can save a lot of space
+* Bad, because serializing and deserialising could introuce errors in the data
+* … <!-- numbers of pros and cons can vary -->
```

Comment:
```suggestion
* Bad, because we need to additionally process the serialization / deserialization - with another processing microservice.
* … <!-- numbers of pros and cons can vary -->
```